Eevee.Sleep.Bot/Utils/DiscordMessageMaker.cs

```diff
@@ -194,6 +194,48 @@ ButtonId buttonId
         return builder.Build();
     }
 
+    public static Embed MakeChangeRoleResult(
+        IUser user,
+        ulong[] previousRoleIds,
+        ulong[] currentRoleIds,
+        Color color
+    ) {
+        return new EmbedBuilder()
+            .WithColor(color)
+            .WithAuthor(user)
+            .WithTitle("Your owned roles")
+            .AddField(
+                name: "Roles before change",
+                value: previousRoleIds.Length == 0 ?
+                    "(N/A)" :
+                    string.Join(" / ", previousRoleIds.Select(x => MentionUtils.MentionRole(x)))
+            )
+            .AddField(
+                name: "Roles after change",
+                value: currentRoleIds.Length == 0 ?
+                    "(N/A)" :
+                    string.Join(" / ", currentRoleIds.Select(x => MentionUtils.MentionRole(x)))
+            )
+            .WithCurrentTimestamp()
+            .Build();
+    }
+
+    public static Embed MakeShowRoleResult(IUser user, ulong[] roleIds) {
+        return new EmbedBuilder()
+            .WithColor(Colors.Success)
+            .WithAuthor(user)
+            .WithTitle("Your owned roles")
+            .AddField(
+                name: "Roles",
+                value: roleIds.Length == 0 ?
```

Comment:
Refactor all `string.Join()` calls to `DiscordExtension.MentionAllRoles()` and call it from there.

Reply:
And I think all `string.Join("\n")` can be further refactored to `StringExtension.mergeLines()` as well